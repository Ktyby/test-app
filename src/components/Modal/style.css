.overlay {
  position: fixed;
  z-index: 999;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;

  background-color: rgba(0, 0, 0, 0.4);
}

.modal {
  position: fixed;
  left: 50%;
  top: 50%;

  transform: translate(-50%, max(-50%, -50vh));

  width: fit-content;
  max-height: 100vh;
  border-radius: 5px;
  padding: 10px;

  background-color: #ffffff;
  box-shadow: 0px 0px 19px 8px rgb(0 0 0 / 20%);

  overflow-y: auto;
}

.modal__heading {
  display: flex;
  justify-content: space-between;

  margin-bottom: 15px;

  font-size: 18px;
}

.modal__title {
  width: 100%;
  margin: 0 0 0 30px;

  text-align: center;
}

.modal__close {
  border: none;

  font-size: 20px;
  color: #114b5f;

  background: none;

  cursor: pointer;
  transition: 0.3s;
}

.modal__close:hover {
  opacity: 0.8;
}

.modal__close:active {
  opacity: 0.6;
}
