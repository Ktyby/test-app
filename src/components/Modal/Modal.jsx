import React, { memo, useCallback } from "react";
import PropTypes from "prop-types";

import "./style.css";

const Modal = memo(({ children, onClose, title, ...props }) => {
  const handleOverlayClick = useCallback((evt) => {
    if (evt.target.classList.contains("overlay")) {
      onClose();
    }
  }, []);

  return (
    <div className="overlay" onClick={handleOverlayClick}>
      <div {...props} className="modal">
        <div className="modal__heading">
          <h3 className="modal__title">{title}</h3>
          <button className="modal__close" onClick={onClose}>
            â•³
          </button>
        </div>
        {children}
      </div>
    </div>
  );
});

Modal.propTypes = {
  children: PropTypes.any,
  onClose: PropTypes.func,
  title: PropTypes.string,
};
Modal.defaultProps = {
  onClose: () => {},
  title: "",
};

export default Modal;
