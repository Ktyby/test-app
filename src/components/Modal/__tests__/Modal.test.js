import renderer from "react-test-renderer";
import React from "react";

import Modal from "../Modal";

describe("components => Modal", () => {
  it("should render Modal component", () => {
    const component = renderer.create(<Modal children={[]} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
