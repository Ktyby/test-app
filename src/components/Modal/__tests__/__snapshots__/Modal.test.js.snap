// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components => Modal should render Modal component 1`] = `
<div
  className="overlay"
  onClick={[Function]}
>
  <div
    className="modal"
  >
    <div
      className="modal__heading"
    >
      <h3
        className="modal__title"
      />
      <button
        className="modal__close"
        onClick={[Function]}
      >
        â•³
      </button>
    </div>
  </div>
</div>
`;
